### PR TITLE
Add BALENA_DEVICE_ARCH environment variable for containers

### PR DESCRIPTION
**Pull request by CameronDiver**
Closes: #1232
Change-type: minor
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: src/config/functions.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import * as Bluebird from 'bluebird';
 import * as _ from 'lodash';
+import { fs } from 'mz';
```
Comment (zvin) [inline src/config/functions.ts]: We could use `fs.promises` as we're on node >= 10

File: src/config/functions.ts [hunk 2]
```diff
@@ -32,6 +34,22 @@ export const fnSchema = {
...
+			const data = await fs.readFile(
+				`${constants.rootMountPoint}/resin-boot/device-type.json`,
+				'utf8',
+			);
+			const deviceInfo = JSON.parse(data);
+
+			return deviceInfo.arch;
+		} catch (e) {
+			log.error(`Unable to get architecture: ${e}`);
+			return 'unavailable';
```
Comment (zvin) [inline src/config/functions.ts]: Why not `unknown` as in `schema-type.ts` ?

File: src/device-api/v2.ts [hunk 3]
```diff
@@ -311,23 +310,25 @@ export function createV2Api(router: Router, applications: ApplicationManager) {
...
-		});
+			return res.status(200).json({
+				status: 'success',
+				info: {
+					arch: deviceArch,
+					deviceType,
+				},
+			});
+		} catch (e) {
+			res.status(400).json({
```
Comment (zvin) [inline src/device-api/v2.ts]: `Bad Request` doesn't seem appropriate here.

File: test/04-service.spec.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import * as _ from 'lodash';
+import * as mz from 'mz';
```
Comment (zvin) [inline test/04-service.spec.ts]: This is not used.

File: src/config/functions.ts (hunk 1 above)
Comment (Page-) [reply src/config/functions.ts]: We still have a node 6 device :(

File: src/config/functions.ts (hunk 2 above)
Comment (CameronDiver) [reply src/config/functions.ts]: I was originally thinking it was for different failure modes, but it's not actually possible with the current situation for `unavailable` to be returned. I guess it's fine.

**Pull request merged**